### ISSUE TITLE
Unable to remove container

### ISSUE THREAD

**Issue by anshumanbh** [docker/docker#14361]
Hi guys,

I made a Dockerfile which kind of messed up things I guess.
I tried to delete images and containers that were created by that Dockerfile but I am stuck now where I can't delete a container. The delete commands looks like this:

Error response from daemon: Cannot destroy container fbc4feb707f8: Failed to set container state to RemovalInProgress: Status is already RemovalInProgress
Error: failed to remove containers: [fbc4feb707f8]

When I do docker info, the output looks like this:

Containers: 1
Images: 27
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 32
 Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.18.0-kali3-amd64
Operating System: Kali GNU/Linux 1.1.0 (moto)
CPUs: 4
Total Memory: 9.79 GiB
Name: kali
ID: URLU:WG5F:RJAB:4WHX:JFPW:N3UR:G644:K7EI:QPER:J66Z:3XGY:FW6Q
WARNING: No memory limit support
WARNING: No swap limit support

Any idea what is going on?

**Comment by anshumanbh** [comment]
and, when I do df -h, the output looks like this:

df: `/var/lib/docker/aufs/mnt/fbc4feb707f8fcee1222465f3afb9056438b8dd219d70e5af23d5cdb14a208f8': Resource temporarily unavailable
Filesystem                                              Size  Used Avail Use% Mounted on
rootfs                                                  284G   12G  258G   5% /
udev                                                     10M     0   10M   0% /dev
tmpfs                                                  1003M  528K 1003M   1% /run
/dev/disk/by-uuid/94e32721-cee2-497f-aea6-df7605613df5  284G   12G  258G   5% /
tmpfs                                                   5.0M     0  5.0M   0% /run/lock
tmpfs                                                   4.4G  992K  4.4G   1% /run/shm
cgroup                                                  4.9G     0  4.9G   0% /sys/fs/cgroup
/dev/disk/by-uuid/94e32721-cee2-497f-aea6-df7605613df5  284G   12G  258G   5% /var/lib/docker/aufs

**Comment by anshumanbh** [comment]
docker version:
Client version: 1.7.0
Client API version: 1.19
Go version (client): go1.4.2
Git commit (client): 0baf609
OS/Arch (client): linux/amd64
Server version: 1.7.0
Server API version: 1.19
Go version (server): go1.4.2
Git commit (server): 0baf609
OS/Arch (server): linux/amd64

docker info:
Containers: 1
Images: 27
Storage Driver: aufs
Root Dir: /var/lib/docker/aufs
Backing Filesystem: extfs
Dirs: 32
Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.18.0-kali3-amd64
Operating System: Kali GNU/Linux 1.1.0 (moto)
CPUs: 4
Total Memory: 9.79 GiB
Name: kali
ID: URLU:WG5F:RJAB:4WHX:JFPW:N3UR:G644:K7EI:QPER:J66Z:3XGY:FW6Q
WARNING: No memory limit support
WARNING: No swap limit support

uname -a:
Linux kali 3.18.0-kali3-amd64 #1 SMP Debian 3.18.6-1~kali2 (2015-03-02) x86_64 GNU/Linux

Provide additional environment details (AWS, VirtualBox, physical, etc.):
Running Docker in Kali. Installed Docker from the scratch. It is a VM running on VMware vCloud Air. 

List the steps to reproduce the issue:
1. I made the following Dockerfile:
FROM owasp/zap2docker-stable
MAINTAINER test
RUN apt-get -y update
RUN apt-get -y upgrade
RUN apt-get install -y python-pip
RUN pip install python-owasp-zap-v2
CMD curl https://gist.githubusercontent.com/anshumanbh/test.py | python

2. the step where apt-get -y upgrade started running stopped after a while. So, I killed the building process. There were a bunch of Docker images and containers created because of the above Dockerfile. I tried listing them and removing them manually. Some commands didn't work and the terminal would just hang. So, I had to abort those as well. Eventually, what ended up being is that I couldn't delete a container and my diskspace seems to have filled a lot.

**Comment by anshumanbh** [comment]
I rebooted my VM. the container appeared "Dead" and I could then remove it. Things look back to normal now. Apologies for jumping the gun. Hopefully, this helps others as well..

**Comment by fermayo** [comment]
This is happening to us at Tutum as well on nodes with 1.7.1:
```
500 Server Error: Internal Server Error ("Cannot destroy container <id>: Failed to set container state to RemovalInProgress: Status is already RemovalInProgress")
```

**Comment by adamtheturtle** [comment]
It seems like https://github.com/docker/docker/commit/c4e49d10143178c718178ff7ce857f4f8ed46a0b will fix this in Docker 1.9.0.
